code/main/media/mediaSession.brs

```diff
@@ -194,49 +193,55 @@ function _adb_MediaSession(clientSessionId as string, configurationModule as obj
-            ' EdgeRequestQueue.process and handle the responses
-            responses = m._edgeRequestQueue.processRequests()
-            ' the responses may include sessionStart response and media event response
-            for each edgeResponse in responses
-                if _adb_isEdgeResponse(edgeResponse) then
-                    try
-                        requestId = edgeResponse.getRequestId()
-                        responseCode = edgeResponse.getResponseCode()
-                        responseString = edgeResponse.getResponseString()
-
-                        ''' only handle the response for sessionStart event
-                        if m._sessionStartHit.requestId = requestId then
-                            ''' Use constants
-                            if responseCode >= m._RESPONSE_CODE_200 and responseCode < m._RESPONSE_CODE_300
-
-                                responseObj = ParseJson(responseString)
-
-                                ''' process the response handles
-                                if not _adb_isEmptyOrInvalidArray(responseObj.handle) then
-                                    m._processEdgeResponseHandles(responseObj.handle)
-                                end if
-
-                                ''' process the error responses
-                                if not _adb_isEmptyOrInvalidArray(responseObj.errors) then
-                                    m._processEdgeResponseErrors(responseObj.errors)
-                                end if
-                            else
-                                ''' Should execute this code when there is a non-recoverable error for sessionStart request
-                                ''' Abort the session
-                                m.close(true)
-                                _adb_logWarning("MediaSession::_processEdgeRequestQueue() - SessionStart request failed with unrecoverable error.")
+        ''' Handle the edge responses for the media events
+        processEdgeResponse: function(responseEvent as object) as void
+            if _adb_isEdgeResponseEvent(responseEvent) then
+                try
+                    requestId = responseEvent.parentId
+                    if requestId = invalid then
+                        _adb_logWarning("MediaSession::processEdgeResponse() - Invalid requestId in the edge response.")
                         return
+                    end if
+
+                    eventData = responseEvent.data
+                    if eventData = invalid then
+                        _adb_logWarning("MediaSession::processEdgeResponse() - Invalid eventData in the edge response.")
+                        return
+                    end if
+
+                    responseCode = eventData.code
+                    responseString = eventData.message
+
+                    ''' only handle the response for sessionStart event
+                    if m._sessionStartHit.requestId = requestId then
```

Comment:
Can you move this check to the top before reading eventData, responseCode and responseString and only handle relevant events?